state-chain/custom-rpc/src/lib.rs

```diff
@@ -280,6 +280,22 @@ pub struct RpcEnvironment {
 	pools: PoolsEnvironment,
 }
 
+#[derive(Serialize, Deserialize, Clone, PartialEq, Eq, Debug)]
+pub struct PoolPriceV2 {
+	pub from_asset: RpcAsset,
+	pub to_asset: RpcAsset,
+	pub price: PoolPrice,
+}
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, Eq, Debug)]
+pub struct RpcPrewitnessedSwap {
+	pub from_asset: RpcAsset,
+	pub to_asset: RpcAsset,
```

Comment:
Also I suggest using base, quote and side. Instead of having the order implicit based on which way around the assets are.

Reply:
Also renamed a bunch of "from"s to "base" and "to"s to "quote"